crates/pathfinder/src/storage/schema/revision_0020.rs

```diff
@@ -0,0 +1,560 @@
+        pub version: TransactionVersion,
+    }
+
+    fn transaction_version_zero() -> TransactionVersion {
+        TransactionVersion(web3::types::H256::zero())
+    }
+
+    /// Represents deserialized L2 deploy transaction data.
+    #[serde_as]
+    #[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+    #[serde(deny_unknown_fields)]
+    pub struct DeployTransaction {
+        pub contract_address: ContractAddress,
+        pub contract_address_salt: ContractAddressSalt,
+        pub class_hash: ClassHash,
+        #[serde_as(as = "Vec<ConstructorParamAsDecimalStr>")]
+        pub constructor_calldata: Vec<ConstructorParam>,
+        pub transaction_hash: StarknetTransactionHash,
+        #[serde_as(as = "TransactionVersionAsHexStr")]
+        #[serde(default = "transaction_version_zero")]
+        pub version: TransactionVersion,
+    }
+
+    #[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+    #[serde(tag = "version")]
+    pub enum InvokeTransaction {
+        #[serde(rename = "0x0")]
+        V0(InvokeTransactionV0),
+        #[serde(rename = "0x1")]
+        V1(InvokeTransactionV1),
+    }
+
+    impl<'de> Deserialize<'de> for InvokeTransaction {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: serde::Deserializer<'de>,
+        {
+            use serde::de;
+            use web3::types::H256;
+
+            #[serde_as]
+            #[derive(Deserialize)]
+            struct Version {
+                #[serde_as(as = "TransactionVersionAsHexStr")]
+                #[serde(default = "transaction_version_zero")]
+                pub version: TransactionVersion,
+            }
+
+            let mut v = serde_json::Value::deserialize(deserializer)?;
+            let version = Version::deserialize(&v).map_err(de::Error::custom)?;
+            // remove "version", since v0 and v1 transactions use deny_unknown_fields
+            v.as_object_mut()
+                .expect("must be an object because deserializing version succeeded")
+                .remove("version");
+            match version.version {
+                TransactionVersion(x) if x == H256::from_low_u64_be(0) => Ok(Self::V0(
+                    InvokeTransactionV0::deserialize(&v).map_err(de::Error::custom)?,
+                )),
+                TransactionVersion(x) if x == H256::from_low_u64_be(1) => {
+                    // Starknet 0.10.0 still has `entry_point_selector` and `entry_point_type` because
+                    // of a bug that will be fixed in 0.10.1. We should just ignore these fields until
+                    // this gets fixed.
+                    // FIXME: 0.10.1 remove this hack
+                    let o = v
+                        .as_object_mut()
+                        .expect("must be an object because deserializing version succeeded");
+                    o.remove("entry_point_selector");
+                    o.remove("entry_point_type");
```

Comment:
What's the point of having a pending removal within a migration...? Is it to serve as a reminder to do another migration when 0.10.1 is released?

Reply:
You're right, it makes no sense here.

This has been copy-pasted over from the sequencer code where it does make sense. I'll remove this FIXME.